// primitives/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = exports.Text = exports.View = exports.createImageComponent = exports.createTextComponent = exports.createViewComponent = exports.StyleSheet = undefined;

var _StyleSheet = require('./web/StyleSheet');

var _StyleSheet2 = _interopRequireDefault(_StyleSheet);

var _createViewComponent = require('./web/createViewComponent');

var _createViewComponent2 = _interopRequireDefault(_createViewComponent);

var _createTextComponent = require('./web/createTextComponent');

var _createTextComponent2 = _interopRequireDefault(_createTextComponent);

var _createImageComponent = require('./web/createImageComponent');

var _createImageComponent2 = _interopRequireDefault(_createImageComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.StyleSheet = _StyleSheet2.default;
exports.createViewComponent = _createViewComponent2.default;
exports.createTextComponent = _createTextComponent2.default;
exports.createImageComponent = _createImageComponent2.default;
var View = exports.View = (0, _createViewComponent2.default)(_StyleSheet2.default);
var Text = exports.Text = (0, _createTextComponent2.default)(_StyleSheet2.default);
var Image = exports.Image = (0, _createImageComponent2.default)(_StyleSheet2.default, View);
