// primitives/StyleSheet/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _ReactNativePropRegistry = require('./ReactNativePropRegistry');

var _ReactNativePropRegistry2 = _interopRequireDefault(_ReactNativePropRegistry);

var _mapValues = require('lodash/mapValues');

var _mapValues2 = _interopRequireDefault(_mapValues);

var _flattenStyle = require('../StyleSheet/flattenStyle');

var _flattenStyle2 = _interopRequireDefault(_flattenStyle);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var absoluteFillObject = {
  position: 'absolute',
  left: 0,
  right: 0,
  top: 0,
  bottom: 0
};

var absoluteFill = _ReactNativePropRegistry2.default.register(absoluteFillObject);

function create(obj) {
  return (0, _mapValues2.default)(obj, function (val) {
    return _ReactNativePropRegistry2.default.register(val);
  });
}

exports.default = {
  absoluteFillObject: absoluteFillObject,
  absoluteFill: absoluteFill,
  hairlineWidth: 1,
  create: create,
  flatten: _flattenStyle2.default
};
