// primitives/web/createViewComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

exports.default = createViewComponent;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createDOMElement = require('./createDOMElement');

var _createDOMElement2 = _interopRequireDefault(_createDOMElement);

var _DefaultStyles = require('./DefaultStyles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function createViewComponent(StyleSheet) {
  var styles = StyleSheet.create({
    // https://github.com/facebook/css-layout#default-values
    initial: _DefaultStyles.defaultView,
    flexible: {
      flexShrink: 1
    }
  });

  return function (_React$PureComponent) {
    _inherits(View, _React$PureComponent);

    function View() {
      _classCallCheck(this, View);

      return _possibleConstructorReturn(this, (View.__proto__ || Object.getPrototypeOf(View)).apply(this, arguments));
    }

    _createClass(View, [{
      key: 'render',
      value: function render() {
        var _props = this.props;
        var pointerEvents = _props.pointerEvents;
        var style = _props.style;

        var other = _objectWithoutProperties(_props, ['pointerEvents', 'style']);

        var flattenedStyle = StyleSheet.flatten(style) || {};
        var pointerEventsStyle = pointerEvents && { pointerEvents: pointerEvents };

        var props = _extends({}, other, {
          // TODO: change View to display: inline-block if it's child of Text to behave like React Native iOS
          style: [styles.initial, style, flattenedStyle.flex && styles.flexible, // react-native behaviour
          pointerEventsStyle]
        });

        return (0, _createDOMElement2.default)('div', _extends({}, props, StyleSheet.resolve(props)));
      }
    }]);

    return View;
  }(_react2.default.PureComponent);
}
