// primitives/web/createImageComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

exports.default = createImageComponent;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ImageResizeMode = {
  center: 'center',
  contain: 'contain',
  cover: 'cover',
  none: 'none',
  repeat: 'repeat',
  stretch: 'stretch'
};

function createImageComponent(StyleSheet, View) {
  var styles = StyleSheet.create({
    initial: {
      backgroundColor: 'transparent',
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover'
    },
    img: {
      borderWidth: 0,
      height: 'auto',
      maxHeight: '100%',
      maxWidth: '100%',
      opacity: 0
    },
    children: {
      bottom: 0,
      left: 0,
      position: 'absolute',
      right: 0,
      top: 0
    }
  });

  var resizeModeStyles = StyleSheet.create({
    center: {
      backgroundSize: 'auto',
      backgroundPosition: 'center'
    },
    contain: {
      backgroundSize: 'contain'
    },
    cover: {
      backgroundSize: 'cover'
    },
    none: {
      backgroundSize: 'auto'
    },
    repeat: {
      backgroundSize: 'auto',
      backgroundRepeat: 'repeat'
    },
    stretch: {
      backgroundSize: '100% 100%'
    }
  });

  return function (_React$PureComponent) {
    _inherits(Image, _React$PureComponent);

    function Image() {
      _classCallCheck(this, Image);

      return _possibleConstructorReturn(this, (Image.__proto__ || Object.getPrototypeOf(Image)).apply(this, arguments));
    }

    _createClass(Image, [{
      key: 'render',
      value: function render() {
        var _props = this.props;
        var defaultSource = _props.defaultSource;
        var source = _props.source;

        var otherProps = _objectWithoutProperties(_props, ['defaultSource', 'source']);

        var displayImage = resolveAssetSource(defaultSource || source);
        var imageSizeStyle = resolveAssetDimensions( /* !isLoaded ? defaultSource : */source);
        var backgroundImage = displayImage ? 'url("' + displayImage + '")' : null;
        var originalStyle = StyleSheet.flatten(this.props.style);
        var resizeMode = this.props.resizeMode || originalStyle && originalStyle.resizeMode || ImageResizeMode.cover;

        // View doesn't support 'resizeMode' as a style
        if (originalStyle) {
          delete originalStyle.resizeMode;
        }

        return _react2.default.createElement(View, _extends({}, otherProps, {
          accessibilityRole: 'img',
          style: [styles.initial, imageSizeStyle, originalStyle, !!backgroundImage && { backgroundImage: backgroundImage }, resizeModeStyles[resizeMode]]
        }));
      }
    }]);

    return Image;
  }(_react2.default.PureComponent);
}

// Helpers

function resolveAssetDimensions(source) {
  if (source && (typeof source === 'undefined' ? 'undefined' : _typeof(source)) === 'object') {
    var _height = source.height;
    var _width = source.width;

    return { height: _height, width: _width };
  }
}

function resolveAssetSource(source) {
  return (source && (typeof source === 'undefined' ? 'undefined' : _typeof(source)) === 'object' ? source.uri : source) || null;
}
